script/DeployHypMinter.s.sol

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import {Script, console2} from "forge-std/Script.sol";
+import {HypMinter} from "../src/contracts/HypMinter.sol";
+import {AccessManager} from "@openzeppelin/contracts/access/manager/AccessManager.sol";
+import {TransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+/**
+ * @title Deploy HypMinter Script
+ * @dev Script to deploy HypMinter contract with proper initialization
+ *
+ * Usage:
+ * forge script script/DeployHypMinter.s.sol:DeployHypMinter --sig "run()" --rpc-url $RPC_URL --broadcast --verify
+ *
+ * Environment Variables:
+ * - PRIVATE_KEY: Private key for deployment
+ */
+contract DeployHypMinter is Script {
+    // Constructor Constants
+    uint256 constant DISTRIBUTION_DELAY_MAXIMUM = 10 days;
+
+    // Deployment configuration => pass to initialize
+    struct DeployConfig {
+        AccessManager accessManager;
+        uint256 firstRewardTimestamp;
+        uint256 mintAllowedTimestamp;
+        uint256 distributionAllowedTimestamp;
+        uint256 distributionDelay;
+        address operatorRewardsManager;
+    }
+
+    function run() public {
+        run({
+            _accessManager: AccessManager(0x3D079E977d644c914a344Dcb5Ba54dB243Cc4863),
+            _firstRewardTimestamp: 1_752_448_487, // Sun Jul 13 2025 19:14:47 EDT
+            _mintAllowedTimestamp: 1_760_536_800, // Wednesday, October 15, 2025 10:00:00 AM GMT-04:00 DST
+            _distributionAllowedTimestamp: 1_761_141_600, // Wednesday, October 22, 2025 10:00:00 AM GMT-04:00
```

Comment:
I thought we would define these as relative values but okay

Reply:
Updated